src/webviews/login/index.ts

```diff
@@ -141,25 +136,25 @@ export class Login {
    * @param context
    */
   static async LoginToPrevious(name: string, context: vscode.ExtensionContext, reloadServerSettings?: boolean) {
-    const connection = instance.getConnection();
-    if (connection) {
+    const existingConnection = instance.getConnection();
+    if (existingConnection) {
       // If the user is already connected and trying to connect to a different system, disconnect them first
-      if (name !== connection.currentConnectionName) {
-        vscode.window.showInformationMessage(`Disconnecting from ${connection.currentHost}.`);
+      if (name !== existingConnection.currentConnectionName) {
+        vscode.window.showInformationMessage(`Disconnecting from ${existingConnection.currentHost}.`);
         if (!disconnect()) return false;
```

Comment:
We're missing an `await` here.
```suggestion
        if (!await disconnect()) return false;
```